diff --git a/include/foonathan/memory/std_allocator.hpp b/include/foonathan/memory/std_allocator.hpp
index ab15cf3..61a4306 100644
--- a/include/foonathan/memory/std_allocator.hpp
+++ b/include/foonathan/memory/std_allocator.hpp
@@ -224,6 +224,8 @@ namespace foonathan
             template <typename U>
             void destroy(U* p) FOONATHAN_NOEXCEPT
             {
+                // This is to avoid a stupid MSVS 2015 'unreferenced formal parameter' warning
+                (void)p;
                 p->~U();
             }
 
